### PR TITLE
Adding playbook backstage tests

### PR DESCRIPTION
**Pull request by prapti**
Adding a couple of more playbook backstage test with plans to add more tests as we go:
* backstage_spec:
    * Verifying that backstage opens playbook creation page with New Playbook button.
    * Verifying that backstage opens playbook creation page with "Blank Playbook" template option
    * Verifying that backstage opens Incident Response Playbook page from its template option
* playbook_details_spec:
    * Backstage playbook details has the page specific UI components

Screenshots:
Screen Shot 2020-09-29 at 5 43 20 PM
Screen Shot 2020-09-29 at 5 47 49 PM

### CODE REVIEW

File: tests-e2e/cypress/integration/backstage/backstage_spec.js [hunk 1]
```diff
@@ -65,4 +67,55 @@ describe('backstage', () => {
         // * Verify that incidents are shown
         cy.findByTestId('titleIncident').should('be.visible').contains('Incidents');
     });
+    it('opens playbook creation page with New Playbook button', () => {
```
Comment (lieut-data) [inline tests-e2e/cypress/integration/backstage/backstage_spec.js:70]: nit: missing newline separating tests
```suggestion

     it('opens playbook creation page with New Playbook button', () => { 
```

Comment (lieut-data) [reply tests-e2e/cypress/integration/backstage/backstage_spec.js:70]: Related, this test feels like it belongs in the `playbook_list_spec.js`? I had imagined the vanilla `backstage_spec.js` was more about the shared elements, like the navigation header.

File: tests-e2e/cypress/integration/backstage/backstage_spec.js [hunk 2]
```diff
@@ -65,4 +67,55 @@ describe('backstage', () => {
...
+        cy.url().should('include', `playbooks/new?template_title=Incident%20Response%20Playbook`);
+
+        // * Verify the playbook name is 'Incident Response Playbook'
+        cy.get('#playbook-name').should('be.visible').within(() => {
+            cy.findByText('Incident Response Playbook').should('be.visible');
+        });
+    });
 });
+
+function verifyPlaybookCreationPageOpened(template=false) {
```
Comment (lieut-data) [inline tests-e2e/cypress/integration/backstage/backstage_spec.js:110]: What does the `template` parameter do?

File: tests-e2e/cypress/integration/backstage/backstage_spec.js [hunk 3]
```diff
@@ -65,4 +67,55 @@ describe('backstage', () => {
...
+    });
+
+    it('opens Incident Response Playbook page from its template option', () => {
+        // # Open backstage
+        cy.openBackstage();
+
+        // # Click 'Incident Response Playbook'
+        cy.findByText('Incident Response Playbook').should('be.visible').click().wait(TIMEOUTS.TINY);
+
+        // * Verify the page url contains 'Incident Response Playbook'
```
Comment (lieut-data) [inline tests-e2e/cypress/integration/backstage/backstage_spec.js:100]: Can we find a way to use `verifyPlaybookCreationPageOpened`?

File: tests-e2e/cypress/integration/backstage/playbook_details_spec.js [hunk 4]
```diff
@@ -19,4 +22,37 @@ describe('backstage playbook details', () => {
         // * Verify that the user has been redirected to the playbooks not found error page
         cy.url().should('include', '/ad-1/com.mattermost.plugin-incident-response/error?type=playbooks');
     });
+
+    it('has playbook specific UI components', () => {
```
Comment (lieut-data) [inline tests-e2e/cypress/integration/backstage/playbook_details_spec.js:26]: I'm not a fan of this test, specifically -- it seems brittle as compared to a test focusing on /doing/ something, like creating a playbook using these specific UI components.

**Review by lieut-data — CHANGES_REQUESTED** [review]
Thanks, @prapti! A few comments below.

File: tests-e2e/cypress/integration/backstage/playbook_details_spec.js (hunk 4 above)
Comment (prapti) [reply tests-e2e/cypress/integration/backstage/playbook_details_spec.js:26]: I didn't have a very strong feeling about adding a "UI component test" either.. getting rid of it :)

File: tests-e2e/cypress/integration/backstage/backstage_spec.js (hunk 2 above)
Comment (prapti) [reply tests-e2e/cypress/integration/backstage/backstage_spec.js:110]: Nothing in this case, it was a bad cleanup on my part. Updated this function for better use.

File: tests-e2e/cypress/integration/backstage/backstage_spec.js (hunk 1 above)
Comment (prapti) [reply tests-e2e/cypress/integration/backstage/backstage_spec.js:70]: I agree that it does not quite fit in the `backstage_spec`. Maybe I'll move it to a `playbook_creation_spec`.

**Review by lieut-data — APPROVED** [review]
Momentum! Yee haw -- thanks, @prapti :)

**Pull request merged**